PROTO_tests/tests/indexing_test.py

```diff
@@ -0,0 +1,98 @@
+    elif dtype is ak.str_:
+        return ak.random_strings_uniform(1, 16, size=size)
+    return None
+
+
+class TestIndexing:
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", NUM_TYPES + [ak.str_])
+    def test_pdarray_uint_indexing(self, prob_size, dtype):
+        # for every pda in array_dict test indexing with uint array and uint scalar
+        ikeys, ukeys = key_arrays(prob_size)
+        pda = value_array(dtype, prob_size)
+        assert pda[np.uint(2)] == pda[2]
+        assert pda[ukeys].to_list() == pda[ikeys].to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_bool_indexing(self, prob_size):
+        u = value_array(ak.uint64, prob_size)
+        b = value_array(ak.bool, prob_size)
+        assert u[b].to_list() == ak.cast(u, ak.int64)[b].to_list()
+        assert u[b].to_list() == ak.cast(u, ak.bigint)[b].to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("num_types", NUM_TYPES)
+    def test_set_uint(self, prob_size, num_types):
+        test_size = prob_size // 2
+        ikeys, ukeys = key_arrays(prob_size)
+        pda = value_array(num_types, prob_size)
+
+        # set [int] = val with uint key and value
+        pda[np.uint(2)] = np.uint(5)
+        assert pda[np.uint(2)] == pda[2]
+
+        # set [slice] = scalar/pdarray
+        pda[:test_size] = -2
+        assert pda[ukeys].to_list() == pda[ikeys].to_list()
+        pda[:test_size] = ak.cast(ak.arange(test_size), num_types)
+        assert pda[ukeys].to_list() == pda[ikeys].to_list()
+
+        # set [pdarray] = scalar/pdarray with uint key pdarray
+        pda[ak.arange(test_size, dtype=ak.uint64)] = np.uint(3)
+        assert pda[ukeys].to_list() == pda[ikeys].to_list()
+        pda[ak.arange(test_size, dtype=ak.uint64)] = ak.cast(ak.arange(test_size), num_types)
+        assert pda[ukeys].to_list() == pda[ikeys].to_list()
+
+        if num_types == ak.bigint:
+            # bigint specific set [int] = val with uint key and value
+            pda[np.uint(2)] = 2**200
+            assert pda[2] == 2**200
+
+            # bigint specific set [slice] = scalar/pdarray
+            pda[:prob_size] = 2**200
+            assert pda[:prob_size].to_list() == ak.full(prob_size, 2**200, ak.bigint).to_list()
+            pda[:prob_size] = ak.arange(prob_size, dtype=ak.bigint)
+            assert pda[:prob_size].to_list() == ak.arange(prob_size, dtype=ak.uint64).to_list()
+
+            # bigint specific set [pdarray] = scalar/pdarray with uint key pdarray
+            pda[ak.arange(prob_size, dtype=ak.uint64)] = 2**200
+            assert pda[:prob_size].to_list() == ak.full(prob_size, 2**200, ak.bigint).to_list()
+            pda[ak.arange(prob_size)] = ak.arange(prob_size, dtype=ak.bigint)
+            assert pda[:prob_size].to_list() == ak.arange(prob_size, dtype=ak.uint64).to_list()
+
```

Comment:
spoke with @jaketrookman about this offline. We are currently special casing for when `dtype == ak.bigint` to make sure we can set values that are representative of `bigint`. But we should really be testing to make sure we can set value that are representative of every dtype

I think let's create a `value_scalar` func that takes in dtype and returns a representative scalar and use that for the scalar, then use `value_array` for setting using `[slice]` and `[pdarray]`

This will result in less code (since we won't have to special case bigint) and better coverage

Reply:
Updated